README.md

```diff
@@ -76,31 +77,30 @@ transformation code.
 // Toy has literals, variables, basic arithmetic and functions that accept a single argument. 
 
 (define toy_lang
-    (domain
-        (sum expr
-            (lit value::ion)
-            (variable name::symbol)
-            (not expr::expr)
-            (apply op operands::(* expr 2))
-            (call name::symbol argument::expr)
-            (let name::symbol value::expr body::expr)
-            (function var_name::symbol body::expr))
-        (sum op (plus) (minus) (times) (divide))
-))
-
-// Define another type domain which is the same as "toy_lang" but replaces variable names with DeBruijn indices:
-(define toy_lang_nameless
+ (domain
+   (sum operator (plus) (minus) (times) (divide) (modulo))
+    (sum expr
+        (lit value::ion)
+        (variable name::symbol)
+        (not expr::expr)
+        (nary op::operator operands::(* expr 0))
+        (let name::symbol value::expr body::expr)
+        (function var_name::symbol body::expr))))
+
+// Define another type domain which extends "toy_lang" by removing named variables and adding DeBruijn indices:
+(define toy_lang_indexed
   (permute_domain toy_lang
     (with expr
       (exclude variable let)
       (include
-        (variable index::int)
-        (let index::int value::expr body::expr)))))
+        // We are adding the `index` element here.
+        (variable name::symbol index::int)
+        (let name::symbol index::int value::expr body::expr)))))
```

Comment:
As an aside, can `name::symbol index::int` be written using `variable`? Something like:

```
let letVar::variable value::expr body::expr)
```

Reply:
I can see that there's a few cases in which we'd like to be able to do something like that, but it is not currently possible.